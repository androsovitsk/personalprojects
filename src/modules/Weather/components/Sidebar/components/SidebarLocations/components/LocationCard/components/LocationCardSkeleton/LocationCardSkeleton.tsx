import React from 'react'
import CardContent from '@mui/material/CardContent'
import Skeleton from '../../../../../../../../../../ui/components/Skeleton/Skeleton'
import Card from '@mui/material/Card'

const LocationCardSkeleton: React.FC = () => {
  return (
    <Card variant={'outlined'}>
      <CardContent>
        <Skeleton variant='rectangular' height={'1rem'} />
        <Skeleton variant='rectangular' height={'1rem'} />
        <Skeleton
          variant='rectangular'
          height={'1rem'}
          width={'66%'}
          noMargin
        />
      </CardContent>
    </Card>
  )
}

LocationCardSkeleton.displayName = 'LocationCardSkeleton'

export default LocationCardSkeleton
