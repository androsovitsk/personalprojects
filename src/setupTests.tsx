import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import fetchMock from 'fetch-mock'
import TranslationProvider from './ui/components/TranslationProvider/TranslationProvider'
import LanguageTypes from './ui/components/TranslationProvider/types/LanguageTypes'
import resourceBundles from './locale'
import { createTheme, ThemeProvider } from '@mui/material'

beforeEach(() => {
  fetchMock.reset()
  jest.clearAllMocks()
})

const Wrapper: React.FC<React.PropsWithChildren> = ({ children }) => {
  return (
    <ThemeProvider theme={createTheme({})}>
      <TranslationProvider
        defaultLanguage={LanguageTypes.ENGLISH}
        resourceBundles={resourceBundles}
      >
        {children}
      </TranslationProvider>
    </ThemeProvider>
  )
}

export const renderWithProviders = (element: React.ReactElement) =>
  render(element, {
    wrapper: Wrapper
  })
