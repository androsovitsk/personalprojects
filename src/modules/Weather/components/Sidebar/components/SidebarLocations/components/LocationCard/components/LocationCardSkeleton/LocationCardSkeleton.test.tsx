import { renderWithProviders } from '../../../../../../../../../../setupTests'
import LocationCardSkeleton from './LocationCardSkeleton'

describe('LocationCardSkeleton', () => {
  it('should match the snapshot', () => {
    const view = renderWithProviders(<LocationCardSkeleton />)

    expect(view.asFragment()).toMatchSnapshot()
  })
})
