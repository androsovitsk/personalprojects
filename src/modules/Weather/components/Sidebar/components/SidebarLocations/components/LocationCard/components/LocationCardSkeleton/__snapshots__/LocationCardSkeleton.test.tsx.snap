// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCardSkeleton should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="height: 1rem;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
        style="height: 1rem;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1sqh8dc-MuiSkeleton-root"
        style="width: 66%; height: 1rem;"
      />
    </div>
  </div>
</DocumentFragment>
`;
